tools/tpm2_createpolicy.c

```diff
@@ -169,10 +171,18 @@ TPM_RC build_policy(createpolicypcr_ctx *ctx,
     TPM2B_NONCE nonceCaller = { { 0, } };
     nonceCaller.t.size = 0;
 
-    // Start policy session.
-    TPM_RC rval = tpm_session_start_auth_with_params(ctx->sapi_context,
+    // Start policy session: TPM_SE_POLICY or TPM_SE_TRIAL
+    TPM_RC rval = 0;
+    if (!REAL_SESSION) {
+        rval = tpm_session_start_auth_with_params(ctx->sapi_context,
+            &ctx->policy_session, TPM_RH_NULL, 0, TPM_RH_NULL, 0, &nonceCaller,
+            &encryptedSalt, TPM_SE_TRIAL, &symmetric, TPM_ALG_SHA256);        
+    } else {
+        printf("HERE\n");
+        rval = tpm_session_start_auth_with_params(ctx->sapi_context,
             &ctx->policy_session, TPM_RH_NULL, 0, TPM_RH_NULL, 0, &nonceCaller,
-            &encryptedSalt, TPM_SE_TRIAL, &symmetric, TPM_ALG_SHA256);
+            &encryptedSalt, TPM_SE_POLICY, &symmetric, TPM_ALG_SHA256);  
```

Comment:
This could be condensed without the else if.... rather than the bool REAL_SESSION, set session_type (formally REAL_SESSION) to TPM_SE_TRIAL. In the option handling, set it to TPM_SE_POLICY if it is for a real session. Than just pass it to tpm_session_start_auth_with_params().